### DISCUSSION TITLE
ISR: If there are no changes, is the page rebuilt anyway?

### DISCUSSION THREAD

**Discussion by sergioviniciuss** [vercel/next.js#25502] [Help] [answered]
Imagine you have a page using **ISR**. 
A user access this page and the revalidation process is triggered. If there are no changes introduced, will NextJS just **keep the old** static page or will it **generate a new one** anyway?

**Comment by sergioviniciuss** [comment, marked as answer]
I made an experiment today and was able to answer this question: the answer is IT GENERATES A NEW ONE ANYWAY.

the experiment:

1. I built a page using ISR. 
2. I navigated to the static html of this page in the server .next/server/pages/page.html, formatted the code and saved it.
3. Without introducing any changes, I accessed the page in the browser (this would retrigger a new build)
At this point my hypotheses was:
1. If there's no change, maybe it won't do anything with the html file.. so it won't regenerate it as it's not required.

But the result is:
the page was formatted again, which means next is indeed regenerating the page even if no changes were introduced in a CMS or DB whose data would be prefetched.

**Reply by sanzhardanybayev** [reply]
@sergioviniciuss thanks for sharing your experience!  I just got this question
